content/docs/reference/design-docs/train-sim-v3/driver-behavior-modules.en.md

```diff
@@ -3,36 +3,80 @@ title: "Driver behavior modules"
 weight: 30
 ---
 
-### Driver behavior modules
 
-```mermaid
-flowchart TD
+## Design specs
 
-subgraph BuilderInputs["builder inputs"]
-    SlowdownCoefficient[slowdown coefficient]
-    PathProps[path properties]
-end
+### General pitch
 
-subgraph CallInputs["call inputs"]
-    ConcreteDrivingInstructions[concrete driving instruction]
-    CurrentTrainState[current train state]
-end
+Driver behavior modules are responsible for making driving decisions. 
+It's main responsibility, given the state of the train and an instruction, is to react to the instruction.
+This reaction is expressed as a new train state.
 
-DriverBehaviorModule[driver behavior module]
-Decision[decision]
+To perform this critical task, it needs access to additional context:
 
-BuilderInputs --> DriverBehaviorModule
-CallInputs --> DriverBehaviorModule --> Decision
+- the physical properties of the path, which are used to make coasting decisions, and to model natural forces.
+- a slowdown kind, dictating how time loss is distributed along the path
+- a slowdown coefficient, which is used to adjust how much time is lost
 
-```
+### API
 
-Driver behavior modules are responsible for making driving decisions. It's main responsibility, given the state of the train,
-is to provide a driving instruction (a braking or traction force). To perform this critical task, it needs access to additional context:
+One driver behavior module is instantiated per driving instruction.
```

Comment:
Do we really want to create lots of small driver behavior instances? What component would be in charge of creating them and to select the right kind?

And I don't have any real use case now, but I have a *feeling* that eventually the driver behavior modules will want to manage their own state along the path (on top of the actual train state)

Reply:
That's something we've been wondering as well, but apparently this is the design that originally was picked when writing the proposal. (Maybe could leave this as a TODO?)

Reply:
Indeed, I think that discussions about what model is best may be out of scope

I'll add a mention about this in the warning section